Source/common/Keychain.h

```diff
@@ -0,0 +1,81 @@
+/// Copyright 2025 North Pole Security, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#ifndef SANTA__COMMON__KEYCHAIN_H
+#define SANTA__COMMON__KEYCHAIN_H
+
+#include <Foundation/Foundation.h>
+#include <Security/Security.h>
+
+#include <memory>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+namespace santa {
+
+namespace keychain_utils {
```

Comment:
Is this our first nested namespace? I can't see any others (after a very quick search on my phone). Is there a reason this is in a new namespace?

Reply:
It might be the first. I think it makes a lot of sense to add this and moving forward we should do this more often. I don't have a good reason why I haven't done it previously... These methods are fairly generic sounding and felt better having them scoped within a namespace that better attributed their functionality. Style guide seems to encourage the use of nested namespaces when appropriate. Reckon the alternative is to name these differently:
```
bool IsValidKeychainItemServiceName(NSString *service);
bool IsValidKeychainItemAccountName(NSString *account);
bool IsValidKeychainItemDescription(NSString *description);
```

Reply:
I'm fine with us using namespaces more where it make sense to, my only concern is being able to easily tie namespaces back to files. It might, for example, make sense in this case to have the namespace be `santa::common::keychain`.

Reply:
Reworked based on offline discussion.